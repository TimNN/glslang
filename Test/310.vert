#version 310 es

shared vec4 s;   // ERROR
layout(local_size_x = 2) out;  // ERROR
buffer vec4 v;  // ERROR

layout(location = 2) uniform mat4 x;
layout(location = 3) uniform mat4 y;
layout(location = 2) out mat4 xi;
layout(location = 3) out mat4 yi;  // ERROR, locations conflict with xi

void main()
{
    uvec2 u2;
    u2 = uaddCarry(u2, u2, u2);
    uint u1;
    u1 = usubBorrow(u1, u1, u1);
    uvec4 u4;
    umulExtended(u4, u4, u4, u4);
    ivec4 i4;
    imulExtended(i4, i4, i4, i4);
    int i1;
    i1 = bitfieldExtract(i1, 4, 5);
    uvec3 u3;
    u3 = bitfieldExtract(u3, 4, 5);
    ivec3 i3;
    i3 = bitfieldInsert(i3, i3, 4, 5);
    u1 = bitfieldInsert(u1, u1, 4, 5);
    ivec2 i2;
    i2 = bitfieldReverse(i2);
    u4 = bitfieldReverse(u4);
    i1 = bitCount(i1);
    i3 = bitCount(u3);
    i2 = findLSB(i2);
    i4 = findLSB(u4);
    i1 = findMSB(i1);
    i2 = findMSB(u2);

    vec3 v3;
    v3 = frexp(v3, i3);
    vec2 v2;
    v2 = ldexp(v2, i2);

    mediump vec4 v4;
    u1 = packUnorm4x8(v4);
    u1 = packSnorm4x8(v4);
    v4 = unpackUnorm4x8(u1);
    v4 = unpackSnorm4x8(u1);
}
