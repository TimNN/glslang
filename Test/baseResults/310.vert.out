310.vert
Warning, version 310 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:3: 'shared' : not supported in this stage: vertex
ERROR: 0:4: 'local_size_x' : there is no such layout identifier for this stage taking an assigned value 
ERROR: 0:5: 'buffer' : buffers can be declared only as blocks 
ERROR: 0:10: 'location' : overlapping use of location 3
ERROR: 4 compilation errors.  No code generated.


Shader version: 310
ERROR: node is still EOpNull!
0:12  Function Definition: main( (void)
0:12    Function Parameters: 
0:?     Sequence
0:15      move second child to first child (highp 2-component vector of uint)
0:15        'u2' (highp 2-component vector of uint)
0:15        Function Call: uaddCarry(vu2;vu2;vu2; (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:17      move second child to first child (highp uint)
0:17        'u1' (highp uint)
0:17        Function Call: usubBorrow(u1;u1;u1; (highp uint)
0:17          'u1' (highp uint)
0:17          'u1' (highp uint)
0:17          'u1' (highp uint)
0:19      Function Call: umulExtended(vu4;vu4;vu4;vu4; (void)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:21      Function Call: imulExtended(vi4;vi4;vi4;vi4; (void)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:23      move second child to first child (highp int)
0:23        'i1' (highp int)
0:23        Function Call: bitfieldExtract(i1;i1;i1; (highp int)
0:23          'i1' (highp int)
0:23          Constant:
0:23            4 (const int)
0:23          Constant:
0:23            5 (const int)
0:25      move second child to first child (highp 3-component vector of uint)
0:25        'u3' (highp 3-component vector of uint)
0:25        Function Call: bitfieldExtract(vu3;i1;i1; (highp 3-component vector of uint)
0:25          'u3' (highp 3-component vector of uint)
0:25          Constant:
0:25            4 (const int)
0:25          Constant:
0:25            5 (const int)
0:27      move second child to first child (highp 3-component vector of int)
0:27        'i3' (highp 3-component vector of int)
0:27        Function Call: bitfieldInsert(vi3;vi3;i1;i1; (highp 3-component vector of int)
0:27          'i3' (highp 3-component vector of int)
0:27          'i3' (highp 3-component vector of int)
0:27          Constant:
0:27            4 (const int)
0:27          Constant:
0:27            5 (const int)
0:28      move second child to first child (highp uint)
0:28        'u1' (highp uint)
0:28        Function Call: bitfieldInsert(u1;u1;i1;i1; (highp uint)
0:28          'u1' (highp uint)
0:28          'u1' (highp uint)
0:28          Constant:
0:28            4 (const int)
0:28          Constant:
0:28            5 (const int)
0:30      move second child to first child (highp 2-component vector of int)
0:30        'i2' (highp 2-component vector of int)
0:30        Function Call: bitfieldReverse(vi2; (highp 2-component vector of int)
0:30          'i2' (highp 2-component vector of int)
0:31      move second child to first child (highp 4-component vector of uint)
0:31        'u4' (highp 4-component vector of uint)
0:31        Function Call: bitfieldReverse(vu4; (highp 4-component vector of uint)
0:31          'u4' (highp 4-component vector of uint)
0:32      move second child to first child (highp int)
0:32        'i1' (highp int)
0:32        Function Call: bitCount(i1; (highp int)
0:32          'i1' (highp int)
0:33      move second child to first child (highp 3-component vector of int)
0:33        'i3' (highp 3-component vector of int)
0:33        Function Call: bitCount(vu3; (highp 3-component vector of int)
0:33          'u3' (highp 3-component vector of uint)
0:34      move second child to first child (highp 2-component vector of int)
0:34        'i2' (highp 2-component vector of int)
0:34        Function Call: findLSB(vi2; (highp 2-component vector of int)
0:34          'i2' (highp 2-component vector of int)
0:35      move second child to first child (highp 4-component vector of int)
0:35        'i4' (highp 4-component vector of int)
0:35        Function Call: findLSB(vu4; (highp 4-component vector of int)
0:35          'u4' (highp 4-component vector of uint)
0:36      move second child to first child (highp int)
0:36        'i1' (highp int)
0:36        Function Call: findMSB(i1; (highp int)
0:36          'i1' (highp int)
0:37      move second child to first child (highp 2-component vector of int)
0:37        'i2' (highp 2-component vector of int)
0:37        Function Call: findMSB(vu2; (highp 2-component vector of int)
0:37          'u2' (highp 2-component vector of uint)
0:40      move second child to first child (highp 3-component vector of float)
0:40        'v3' (highp 3-component vector of float)
0:40        Function Call: frexp(vf3;vi3; (highp 3-component vector of float)
0:40          'v3' (highp 3-component vector of float)
0:40          'i3' (highp 3-component vector of int)
0:42      move second child to first child (highp 2-component vector of float)
0:42        'v2' (highp 2-component vector of float)
0:42        Function Call: ldexp(vf2;vi2; (highp 2-component vector of float)
0:42          'v2' (highp 2-component vector of float)
0:42          'i2' (highp 2-component vector of int)
0:45      move second child to first child (highp uint)
0:45        'u1' (highp uint)
0:45        Function Call: packUnorm4x8(vf4; (highp uint)
0:45          'v4' (mediump 4-component vector of float)
0:46      move second child to first child (highp uint)
0:46        'u1' (highp uint)
0:46        Function Call: packSnorm4x8(vf4; (highp uint)
0:46          'v4' (mediump 4-component vector of float)
0:47      move second child to first child (mediump 4-component vector of float)
0:47        'v4' (mediump 4-component vector of float)
0:47        Function Call: unpackUnorm4x8(u1; (mediump 4-component vector of float)
0:47          'u1' (highp uint)
0:48      move second child to first child (mediump 4-component vector of float)
0:48        'v4' (mediump 4-component vector of float)
0:48        Function Call: unpackSnorm4x8(u1; (mediump 4-component vector of float)
0:48          'u1' (highp uint)
0:?   Linker Objects
0:?     's' (shared highp 4-component vector of float)
0:?     'v' (buffer highp 4-component vector of float)
0:?     'x' (layout(location=2 ) uniform highp 4X4 matrix of float)
0:?     'y' (layout(location=3 ) uniform highp 4X4 matrix of float)
0:?     'xi' (layout(location=2 ) smooth out highp 4X4 matrix of float)
0:?     'yi' (layout(location=3 ) smooth out highp 4X4 matrix of float)
0:?     'gl_VertexID' (gl_VertexId highp int)
0:?     'gl_InstanceID' (gl_InstanceId highp int)


Linked vertex stage:


Shader version: 310
ERROR: node is still EOpNull!
0:12  Function Definition: main( (void)
0:12    Function Parameters: 
0:?     Sequence
0:15      move second child to first child (highp 2-component vector of uint)
0:15        'u2' (highp 2-component vector of uint)
0:15        Function Call: uaddCarry(vu2;vu2;vu2; (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:15          'u2' (highp 2-component vector of uint)
0:17      move second child to first child (highp uint)
0:17        'u1' (highp uint)
0:17        Function Call: usubBorrow(u1;u1;u1; (highp uint)
0:17          'u1' (highp uint)
0:17          'u1' (highp uint)
0:17          'u1' (highp uint)
0:19      Function Call: umulExtended(vu4;vu4;vu4;vu4; (void)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:19        'u4' (highp 4-component vector of uint)
0:21      Function Call: imulExtended(vi4;vi4;vi4;vi4; (void)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:21        'i4' (highp 4-component vector of int)
0:23      move second child to first child (highp int)
0:23        'i1' (highp int)
0:23        Function Call: bitfieldExtract(i1;i1;i1; (highp int)
0:23          'i1' (highp int)
0:23          Constant:
0:23            4 (const int)
0:23          Constant:
0:23            5 (const int)
0:25      move second child to first child (highp 3-component vector of uint)
0:25        'u3' (highp 3-component vector of uint)
0:25        Function Call: bitfieldExtract(vu3;i1;i1; (highp 3-component vector of uint)
0:25          'u3' (highp 3-component vector of uint)
0:25          Constant:
0:25            4 (const int)
0:25          Constant:
0:25            5 (const int)
0:27      move second child to first child (highp 3-component vector of int)
0:27        'i3' (highp 3-component vector of int)
0:27        Function Call: bitfieldInsert(vi3;vi3;i1;i1; (highp 3-component vector of int)
0:27          'i3' (highp 3-component vector of int)
0:27          'i3' (highp 3-component vector of int)
0:27          Constant:
0:27            4 (const int)
0:27          Constant:
0:27            5 (const int)
0:28      move second child to first child (highp uint)
0:28        'u1' (highp uint)
0:28        Function Call: bitfieldInsert(u1;u1;i1;i1; (highp uint)
0:28          'u1' (highp uint)
0:28          'u1' (highp uint)
0:28          Constant:
0:28            4 (const int)
0:28          Constant:
0:28            5 (const int)
0:30      move second child to first child (highp 2-component vector of int)
0:30        'i2' (highp 2-component vector of int)
0:30        Function Call: bitfieldReverse(vi2; (highp 2-component vector of int)
0:30          'i2' (highp 2-component vector of int)
0:31      move second child to first child (highp 4-component vector of uint)
0:31        'u4' (highp 4-component vector of uint)
0:31        Function Call: bitfieldReverse(vu4; (highp 4-component vector of uint)
0:31          'u4' (highp 4-component vector of uint)
0:32      move second child to first child (highp int)
0:32        'i1' (highp int)
0:32        Function Call: bitCount(i1; (highp int)
0:32          'i1' (highp int)
0:33      move second child to first child (highp 3-component vector of int)
0:33        'i3' (highp 3-component vector of int)
0:33        Function Call: bitCount(vu3; (highp 3-component vector of int)
0:33          'u3' (highp 3-component vector of uint)
0:34      move second child to first child (highp 2-component vector of int)
0:34        'i2' (highp 2-component vector of int)
0:34        Function Call: findLSB(vi2; (highp 2-component vector of int)
0:34          'i2' (highp 2-component vector of int)
0:35      move second child to first child (highp 4-component vector of int)
0:35        'i4' (highp 4-component vector of int)
0:35        Function Call: findLSB(vu4; (highp 4-component vector of int)
0:35          'u4' (highp 4-component vector of uint)
0:36      move second child to first child (highp int)
0:36        'i1' (highp int)
0:36        Function Call: findMSB(i1; (highp int)
0:36          'i1' (highp int)
0:37      move second child to first child (highp 2-component vector of int)
0:37        'i2' (highp 2-component vector of int)
0:37        Function Call: findMSB(vu2; (highp 2-component vector of int)
0:37          'u2' (highp 2-component vector of uint)
0:40      move second child to first child (highp 3-component vector of float)
0:40        'v3' (highp 3-component vector of float)
0:40        Function Call: frexp(vf3;vi3; (highp 3-component vector of float)
0:40          'v3' (highp 3-component vector of float)
0:40          'i3' (highp 3-component vector of int)
0:42      move second child to first child (highp 2-component vector of float)
0:42        'v2' (highp 2-component vector of float)
0:42        Function Call: ldexp(vf2;vi2; (highp 2-component vector of float)
0:42          'v2' (highp 2-component vector of float)
0:42          'i2' (highp 2-component vector of int)
0:45      move second child to first child (highp uint)
0:45        'u1' (highp uint)
0:45        Function Call: packUnorm4x8(vf4; (highp uint)
0:45          'v4' (mediump 4-component vector of float)
0:46      move second child to first child (highp uint)
0:46        'u1' (highp uint)
0:46        Function Call: packSnorm4x8(vf4; (highp uint)
0:46          'v4' (mediump 4-component vector of float)
0:47      move second child to first child (mediump 4-component vector of float)
0:47        'v4' (mediump 4-component vector of float)
0:47        Function Call: unpackUnorm4x8(u1; (mediump 4-component vector of float)
0:47          'u1' (highp uint)
0:48      move second child to first child (mediump 4-component vector of float)
0:48        'v4' (mediump 4-component vector of float)
0:48        Function Call: unpackSnorm4x8(u1; (mediump 4-component vector of float)
0:48          'u1' (highp uint)
0:?   Linker Objects
0:?     's' (shared highp 4-component vector of float)
0:?     'v' (buffer highp 4-component vector of float)
0:?     'x' (layout(location=2 ) uniform highp 4X4 matrix of float)
0:?     'y' (layout(location=3 ) uniform highp 4X4 matrix of float)
0:?     'xi' (layout(location=2 ) smooth out highp 4X4 matrix of float)
0:?     'yi' (layout(location=3 ) smooth out highp 4X4 matrix of float)
0:?     'gl_VertexID' (gl_VertexId highp int)
0:?     'gl_InstanceID' (gl_InstanceId highp int)

